process/block/shardblock.go

```diff
@@ -46,7 +47,9 @@ type shardProcessor struct {
 
 // SetAppStatusHandler method is used to set appStatusHandler
 func (sp *shardProcessor) SetAppStatusHandler(handler core.AppStatusHandler) {
```

Comment:
The pattern already used in app status handlers impl is that this func should return error if the provided handler is nil. Let's keep this pattern.

Reply:
SetAppStatusHandler return error now